agent/internal/taskoutput/test_log.go

```diff
@@ -151,7 +152,9 @@ func (h *testLogDirectoryHandler) run(ctx context.Context) error {
 	var wg sync.WaitGroup
 	// use runtime.GOMAXPROCS(0) instead of runtime.NumCPU() so that we get the number of CPUs available to Evergreen, not the number of CPUs on the node.
 	// The max procs is set to the actual number of CPUs by automaxprocs in startWebService.
-	for i := 0; i < runtime.GOMAXPROCS(0); i++ {
+	cpus := float64(runtime.GOMAXPROCS(0))
```

Comment:
Can you please elaborate a bit more on why you think that the  root cause is that

> the heartbeat goroutine gets snuffed out by log goroutines doing processing for 7 minutes straight?

Reply:
I agree that using  runtime.Gosched is a better a approach. It'll allow other goroutines to run without suspending the current goroutine and without potentially limiting throughput unnecessarily (which may have happened with the other approach). 
I would suggest monitoring this after deploying it to see if it fixes the bug before closing the ticket.